### PR TITLE
Adjust boot image creation with repos on sle 15

### PR DESCRIPTION
**Pull request by rwx788**
We already have a method to workaround broken patterns, whereas in some
scenarios we have installation_overview_before where we just assert
overview screen.
As a temporary solution we will add repos manually before we get scc.
Whereas, it should not affect test code, as this part is done using test
suite configuration. This PR mainly contains changes to workaround known
installation issues. We also exclude release notes test on sle 15 as it doesn't
work properly with addons and there will be no such a thing visible in
future.

See [poo#20824](https://progress.opensuse.org/issues/20824) and [poo#20984](https://progress.opensuse.org/issues/20984)

NOTE: all workaround take place only on sle15, hence won't affect other versions.

This change is partly included in [PR#3353](https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/3353).
Verification runs: 
[SLE15 minimal base + SDK](http://10.160.66.147/tests/1779)
[SLE15 text mode create hdd](http://10.160.66.147/tests/1769)
[SLE12 SP3 minimal base + SDK](http://10.160.66.147/tests/1731)

[Needles MR#442](https://gitlab.suse.de/openqa/os-autoinst-needles-sles/merge_requests/442) - already merged
[Needles MR#443](https://gitlab.suse.de/openqa/os-autoinst-needles-sles/merge_requests/443) - needs merge.

### CODE REVIEW

File: lib/y2logsstep.pm [hunk 1]
```diff
@@ -164,6 +165,15 @@ sub deal_with_dependency_issues {
     }
 }
 
+sub sle15_workaround_broken_patterns {
+    return unless sle_version_at_least('15');
+    # SLE 15 has pattern errors, workaround them - rbrown 04/07/2017
+    while (check_screen('sle-15-failed-to-select-pattern')) {
```
Comment (okurz) [inline lib/y2logsstep.pm]: why did you change the timeout? This will now cause a lot of waiting time, 30 seconds default. Do you think less could work?

Comment (rwx788) [inline lib/y2logsstep.pm]: My bad, missed that when was resolving a conflict. Good catch.

**Comment by rwx788** [comment]
@okurz If we don't want to wait, we can merge this PR and see where test will fail afterwards.

**Comment by rwx788** [comment]
Additional changes will be required as after resolving package conflicts we get back to overview page and get same error again.

File: tests/installation/grub_test.pm [hunk 2]
```diff
@@ -59,7 +59,7 @@ sub run {
     }
     workaround_type_encrypted_passphrase;
     # 60 due to rare slowness e.g. multipath poo#11908
-    assert_screen "grub2", 60;
+    assert_screen "grub2", 300;
```
Comment (rwx788) [inline tests/installation/grub_test.pm]: Hack, as somehow system boots to the iso after reboot. With increased timeout we start booting from hdd and everything works well.

Comment (okurz) [inline tests/installation/grub_test.pm]: Are you sure this also happened in production? It will happen when you do tests locally with loading from qemu screenshots so that you never finish the normal installation from start to end. Then your machine boots from DVD as well then.

Comment (rwx788) [inline tests/installation/grub_test.pm]: This may explain it. I'll do full verification run and see if anything needed here.

**Comment by rwx788** [comment]
Set back to WIP to address issue with REPO and ADDON_URL variables. To have single test suite for SLE 12 and SLE 15 we need to set variables correctly in main.pm. Blocked on [this merge request](https://gitlab.suse.de/openqa/scripts/merge_requests/107)

**Comment by okurz** [comment]
+1

File: tests/installation/addon_products_sle.pm [hunk 3]
```diff
@@ -51,7 +51,7 @@ sub run {
             # might involve some network lookup of products, licenses, etc.
             assert_screen 'addon-products', 90;
             send_key "tab";                                                     # select addon-products-$addon
-            wait_still_screen 2;
+            wait_still_screen 10;
```
Comment (rwx788) [inline tests/installation/addon_products_sle.pm]: In my local setup list is shown as empty first, then gets to initializing and then proper content is shown. Otherwise, start navigating in menu without a change to reach the list

File: tests/installation/addon_products_sle.pm [hunk 4]
```diff
@@ -96,6 +96,7 @@ sub run {
                 send_key 'alt-t';
             }
             send_key "tab";                                                     # select addon-products-$addon
+            wait_still_screen 10;                                               # wait until repo is added and list is initialized
```
Comment (rwx788) [inline tests/installation/addon_products_sle.pm]: Same as above.
